build.sh

```diff
@@ -1,5 +1,21 @@
 #!/bin/sh
 
+while getopts "d" flag; do
+ case $flag in
+   d) # Handle the -d flag
+   # run docker
+   echo "Running in Docker..."
+    # run docker
+    docker run --rm -it -v `pwd`:/app --name sencha ghcr.io/bwbohl/sencha-cmd:latest /bin/bash -c "./build.sh testing; exit"
```

Comment:
instead of hard coding the "testing" flag I suggest to read this from the initial command line arguments and pass it on to the Docker command
```suggestion
    shift $((OPTIND-1)) # remove -d from the list of input arguments
    OPTIONS=${@} # set a variable OPTIONS with the remaining input arguments to pass to the build command.
    # run docker
    docker run --rm -it -v `pwd`:/app --name sencha ghcr.io/bwbohl/sencha-cmd:latest ./build.sh $OPTIONS"
```

With that you can also do e.g. `./build.sh -d production`

Additionally, I removed the `exit` which is not needed imho

Reply:
With this I get:
```
~/git-clones/Edirom/Edirom-Online$ sh build.sh -d
build.sh: 37: Syntax error: Unterminated quoted string
```
Yes right, the `exit` is not needed because the container is closed when the script finishes.

Reply:
Tested again and it works for me. For better readability I paste the full file:
```sh
#!/bin/sh

while getopts "d" flag; do
 case $flag in
   d) # Handle the -d flag
   # run docker
   echo "Running in Docker..."
    # run docker
    shift $((OPTIND-1)) # remove -d from the list of input arguments
    OPTIONS=${@} # set a variable OPTIONS with the remaining input arguments to pass to the build command
    docker run --rm -it -v `pwd`:/app --name sencha ghcr.io/bwbohl/sencha-cmd:latest ./build.sh $OPTIONS
   exit
   ;;   
   \?)
   # Handle invalid options
   ;;
 esac
done


# cleaning the build dir
sencha ant clean

# building the app
sencha app build $1

# get additional stuff for exist-db
ant build-plus

# build xar
ant xar
```

Reply:
Yes, thank you. It works